src/Type/Php/ArrayMergeFunctionDynamicReturnTypeExtension.php

```diff
@@ -42,12 +44,24 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 
 			$keyTypes[] = TypeUtils::generalizeType($argType->getIterableKeyType(), GeneralizePrecision::moreSpecific());
 			$valueTypes[] = $argType->getIterableValueType();
+
+			if (!$argType->isIterableAtLeastOnce()->yes()) {
+				continue;
+			}
+
+			$nonEmpty = true;
```

Comment:
this doesn't make much sense to me in this case, but phpcbf enforced the early-continue construct